File: united_we_stand/united.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int case_num;
    cin >> case_num;
    // Read cases
    for (int i = 0; i < case_num; i++)
    {
        // Stores array a
        int n;
        cin >> n;
        int *arr = new int[n];
        int max = 0;
        for (int j = 0; j < n; j++)
        {
            cin >> arr[j];
            if (arr[j] > max)
            {
                max = arr[j];
            }
        }

        // Count the number of max numbers
        int count_max = count(arr, arr + n, max);
        int count_others = n - count_max;

        // If the whole array is made up of equal numbers, then it is not possible to created the required arrays
        if (count_max == n)
        {
            cout << -1 << "\n";
        }

        else
        {
            cout << count_others << " " << count_max << "\n";
            for (int j = 0; j < n; j++)
            {
                if (arr[j] != max)
                {
                    cout << arr[j] << " ";
                }
            }

            cout << "\n";

            for (int j = 0; j < count_max; j++)
            {
                cout << max << " ";
            }

            cout << "\n";
        }

        delete[] arr;
    }
}

File: united_we_stand/united.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        n = int(input())
        array = input().split()
        arr = [int(i) for i in array]
        max_num = max(arr)
        count_max = arr.count(max_num)
        count_rest = n - count_max
        if count_max == n:
            print(-1)
        else:
            print(count_rest, count_max)
            for element in arr:
                if element != max_num:
                    print(element, end = " ")
            print()
            for i in range(count_max):
                print(max_num, end = " ")
            print()
            

if __name__ == "__main__":
    main()